scripts/updatekinto.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const DEBUG = true;
+
+const got = require("got");
+
+const SERVER = "https://settings-writer.prod.mozaws.net/v1";
+const CID = "fxmonitor-breaches";
+/* eslint-disable-next-line no-process-env */
+const USERNAME = process.env.KINTO_USERNAME;
+/* eslint-disable-next-line no-process-env */
+const PASSWORD = process.env.KINTO_PASSWORD;
+if (!USERNAME || !PASSWORD) {
+  console.error("Please set credentials in the environment.");
+  return;
+}
+
+const AUTH = Buffer.from(`${USERNAME}:${PASSWORD}`).toString("base64");
+
+async function run() {
+  const RemoteSettingsBreachesSet = new Set((await got(
+    "https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/fxmonitor-breaches/records",
+    { json: true }
+  )).body.data.map(b => b.Name));
+
+  const hibp_breaches = (await got(
+    "https://haveibeenpwned.com/api/v2/breaches",
+    { json: true }
+  )).body;
+
+  const new_breaches = [];
+
+  for (const breach of hibp_breaches) {
+    if (breach.IsSpamList || breach.IsRetired || !breach.IsVerified || !breach.Domain) {
+      continue;
+    }
+
+    if (RemoteSettingsBreachesSet.has(breach.Name)) {
+      continue;
+    }
+
+    new_breaches.push(breach);
```

Comment:
We're pushing the original HIBP breach object into the `new_breaches` array here, and then seemingly using a subset of the fields below on L46-50. Not sure if we should just store the 5-8 keys we actually care about here:

```js
new_breaches.push({
  Name: breach.Name,
  Domain: breach.Domain,
  BreachDate: breach.BreachDate,
  PwnCount: breach.PwnCount,
  AddedDate: breach.AddedDate,
  // See note below
  LogoPath: breach.LogoPath,
  DataClasses: breach.DataClasses,
  Title: breach.Title
});
```

Reply:
Yup, this would work. I'd like to keep the explicit fields in the post request though, which will prevent us from accidentally sending extra data in the future if we make changes. (just as a human-error safety net)